Xamarin.Forms.Platform.Android/AppCompat/TabbedPageRenderer.cs

```diff
@@ -633,6 +640,42 @@ void UpdateBottomNavigationViewIcons()
 			}
 		}
 
+		void UpdateBottomNavigationShiftMode()
+		{
+			if (IsDisposed || _bottomNavigationView == null || !IsBottomTabPlacement)
+				return;
+			var bottomNavigationView = _bottomNavigationView;
+
+			try
+			{
+				var menuView = bottomNavigationView.GetChildAt(0) as BottomNavigationMenuView;
+				if (menuView == null)
+					return;
+
+				var shiftMode = menuView.Class.GetDeclaredField("mShiftingMode");
```

Comment:
What if this was obfuscated by Proguard? we might need to add some rules to it.

Reply:
One thing to note about this is that with API 28 google is grey listing reflection calls

https://developer.android.com/about/versions/pie/restrictions-non-sdk-interfaces

We have an issue to look at this
https://github.com/xamarin/Xamarin.Forms/issues/4118

It looks like Version 28 of the support libraries is going to support setting this so we'll need to determine if we want to add a grey listed api like this until we can upgrade Xamarin Forms to the newer support libraries

Reply:
[LabelVisibilityMode](https://developer.android.com/reference/com/google/android/material/bottomnavigation/LabelVisibilityMode) has different behaviour than api function ```SetBottomToolbarShiftMode```.
And problems that related with reflection (grey list, Proguard, etc) are pretty unpleasant. 

If we want to wait for new api, we will lose this functionality only for api 26,27 (if I correctly understand how Xamarin forms and support libraries work). 

I think we should wait until Xamarin forms will be upgraded to the newer support libraries.

P.S. And after that I will correct review remarks and suggestions.